src/osd/PG.cc

```diff
@@ -6027,6 +6037,100 @@ void PG::update_store_on_load()
   }
 }
 
+struct C_DeleteMore : public Context {
+  PGRef pg;
+  epoch_t epoch;
+  int count = 2;
+  C_DeleteMore(PG *p, epoch_t e) : pg(p), epoch(e) {}
+  void finish(int r) override {
+    ceph_abort();
+  }
+  void complete(int r) override {
+    // complete will be called exactly count times; only the last time will actualy
+    // complete.
+    if (--count) {
+      return;
```

Comment:
Should we be checking r before returning, in case we get an error the first time through?

Otherwise, 👍!

Reply:
these are always 0.. i'll add an assert!